tests/integ/modin/crosstab/test_crosstab.py

```diff
@@ -0,0 +1,767 @@
+                    a,
+                    [b[c] for c in b.columns],
+                    rownames=["a"],
+                    colnames=["b", "c"],
+                    dropna=dropna,
+                )
+            else:
+                return pd.crosstab(
+                    a, b, rownames=["a"], colnames=["b", "c"], dropna=dropna
+                )
+
+        with SqlCounter(query_count=query_count, join_count=join_count):
+            native_args = [a, b]
+            snow_args = [pd.Series(a), pd.DataFrame(b)]
+            eval_snowpark_pandas_result(
+                snow_args,
+                native_args,
+                eval_func,
+            )
+
+    def test_margins(self, dropna):
+        query_count = 3
+        join_count = 3 if dropna else 6
+        union_count = 3
+        a = np.array(
+            [
+                "foo",
+                "foo",
+                "foo",
+                "foo",
+                "bar",
+                "bar",
+                "bar",
+                "bar",
+                "foo",
+                "foo",
+                "foo",
+            ],
+            dtype=object,
+        )
+        b = np.array(
+            [
+                "one",
+                "one",
+                "one",
+                "two",
+                "one",
+                "one",
+                "one",
+                "two",
+                "two",
+                "two",
+                "one",
+            ],
+            dtype=object,
+        )
+        c = np.array(
+            [
+                "dull",
+                "dull",
+                "shiny",
+                "dull",
+                "dull",
+                "shiny",
+                "shiny",
+                "dull",
+                "shiny",
+                "shiny",
+                "shiny",
+            ],
+            dtype=object,
+        )
+        with SqlCounter(
+            query_count=query_count, join_count=join_count, union_count=union_count
+        ):
+            native_df = native_pd.crosstab(
+                a,
+                [b, c],
+                rownames=["a"],
+                colnames=["b", "c"],
+                margins=True,
+                margins_name="MARGINS_NAME",
+                dropna=dropna,
+            )
+            snow_df = pd.crosstab(
+                a,
+                [b, c],
+                rownames=["a"],
+                colnames=["b", "c"],
+                margins=True,
+                margins_name="MARGINS_NAME",
+                dropna=dropna,
+            )
+            assert_snowpark_pandas_equal_to_pandas(snow_df, native_df)
+
+    @pytest.mark.parametrize("normalize", [0, 1, True, "all", "index", "columns"])
+    def test_normalize(self, dropna, normalize):
+        query_count = 4 if normalize not in (0, "index") else 3
+        join_count = 0 if normalize not in (0, "index") else 3
+        if not dropna:
+            join_count = 9 if normalize in (0, "index") else 4
```

Comment:
```suggestion
        query_count = 3 if normalize in (0, "index") else 4
        if dropna:
	    join_count = 3 if normalize in (0, "index") else 0
        else:
            join_count = 9 if normalize in (0, "index") else 4
```

nit: reduce `not`s and only assign `join_count` once.